packages/gestalt/src/TapArea.js

```diff
@@ -45,16 +45,19 @@ type BaseTapArea = {|
   onMouseEnter?: MouseEventHandler,
   onMouseLeave?: MouseEventHandler,
   onTap?: OnTapType,
-  tabIndex?: -1 | 0,
+  role?: 'button' | 'link' | 'switch',
   rounding?: Rounding,
+  tabIndex?: -1 | 0,
   tapStyle?: 'none' | 'compress',
 |};
+
 type TapAreaType = {|
   ...BaseTapArea,
   accessibilityControls?: string,
   accessibilityExpanded?: boolean,
   accessibilityHaspopup?: boolean,
-  role?: 'button',
+  accessibilityPressed?: boolean,
+  role?: 'button' | 'switch',
```

Comment:
For an item with role="switch", you'll also need to supply the aria-checked attribute https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/switch_role

Reply:
Ah, good catch. Just added

Reply:
Update: just _replaced_ 😅